mcstatus/protocol/connection.py

```diff
@@ -221,119 +525,140 @@ def read(self, length: int) -> bytearray:
             result.extend(new)
         return result
 
-    def write(self, data: Union[bytes, bytearray]) -> None:
+    def write(self, data: Union[Connection, str, bytes, bytearray]) -> None:
+        """Send data on self.socket."""
+        if isinstance(data, Connection):
+            data = bytearray(data.flush())
+        elif isinstance(data, str):
+            data = bytearray(data, "utf-8")
         self.socket.send(data)
 
-    def __del__(self):
-        try:
-            self.socket.close()
-        except Exception:  # TODO: Check what this actually excepts
-            pass
 
+class UDPSocketConnection(SocketConnection):
+    """UDP Connection class"""
+
+    __slots__ = ("addr",)
 
-class UDPSocketConnection(Connection):
     def __init__(self, addr: Address, timeout: float = 3):
-        Connection.__init__(self)
+        """Set self.addr to address, set self.socket to new socket,
+        AF_INET if IPv4, AF_INET6 otherwise."""
```

Comment:
Our maximum line length is 127, so this can easily fit into a single docstring line.
```suggestion
        """Set self.addr to address, set self.socket to new socket, AF_INET if IPv4, AF_INET6 otherwise."""
```

Reply:
Gotcha